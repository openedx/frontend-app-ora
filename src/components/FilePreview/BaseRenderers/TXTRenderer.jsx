import React from 'react';
import PropTypes from 'prop-types';
import { rendererHooks } from './textHooks';

const TXTRenderer = ({ url, onError, onSuccess }) => {
  const { content } = rendererHooks({ url, onError, onSuccess });
  return (
    <pre className="txt-renderer">
      {content}
    </pre>
  );
};

TXTRenderer.defaultProps = {};

TXTRenderer.propTypes = {
  url: PropTypes.string.isRequired,
  onError: PropTypes.func.isRequired,
  onSuccess: PropTypes.func.isRequired,
};

export default TXTRenderer;
