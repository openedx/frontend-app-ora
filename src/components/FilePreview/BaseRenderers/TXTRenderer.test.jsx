import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import TXTRenderer from './TXTRenderer';

jest.mock('./textHooks', () => {
  const content = 'test-content';
  return {
    content,
    rendererHooks: (args) => ({ content, rendererHooks: args }),
  };
});

describe('TXT Renderer Component', () => {
  const props = {
    url: 'some_url.txt',
    onError: jest.fn().mockName('this.props.onError'),
    onSuccess: jest.fn().mockName('this.props.onSuccess'),
  };
  test('snapshot', () => {
    expect(shallow(<TXTRenderer {...props} />).snapshot).toMatchSnapshot();
  });
});
