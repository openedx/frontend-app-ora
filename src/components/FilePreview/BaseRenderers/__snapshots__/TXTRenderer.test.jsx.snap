// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TXT Renderer Component snapshot 1`] = `
<pre
  className="txt-renderer"
>
  test-content
</pre>
`;
